### ISSUE TITLE
Tags for Nested Structures Are Not Properly Reflected in OpenAPI Specification Generation

### ISSUE THREAD

**Issue by Ignaciojeria** [go-fuego/fuego#318]
When extracting a structure and using it as an attribute in another structure, the OpenAPI specification generation does not consider the tags for the external structure. Specifically, tags such as example and description for the nested structure (References in this case) are not being included in the generated OpenAPI specification.

Example:
Here is a sample Go package illustrating the issue:

```
package models

type Pets struct {
    ID         string     `json:"id" validate:"required" example:"pet-123456"`
    Name       string     `json:"name" validate:"required" example:"Napoleon"`
    Age        int        `json:"age" example:"2" description:"Age of the pet, in years"`
    IsAdopted  bool       `json:"is_adopted" description:"Is the pet adopted"`
    References References `json:"references"`
}

type PetsCreate struct {
    Name       string     `json:"name" validate:"required,min=1,max=100" example:"Napoleon"`
    Age        int        `json:"age" validate:"min=0,max=100" example:"2" description:"Age of the pet, in years"`
    IsAdopted  bool       `json:"is_adopted" description:"Is the pet adopted"`
    References References `json:"references"`
}

type PetsUpdate struct {
    Name       string     `json:"name,omitempty" validate:"min=1,max=100" example:"Napoleon" description:"Name of the pet"`
    Age        int        `json:"age,omitempty" validate:"max=100" example:"2"`
    IsAdopted  *bool      `json:"is_adopted,omitempty" description:"Is the pet adopted"`
    References References `json:"references"`
}

type References struct {
    Type  string `json:"type" validate:"required" example:"pet-123456"`
    Value string `json:"value"`
}
```

Observed Behavior:
The OpenAPI specification generated does not include the example and description tags for the nested structure References.

Expected Behavior:
The generated OpenAPI specification should reflect all tags (e.g., example and description) for nested structures, ensuring the documentation is complete and accurate.

![image](https://github.com/user-attachments/assets/13bb8382-3e26-4e0f-b196-5f00aabedcef)

**Comment by EwenQuim** [comment]
Well found!

@dylanhitt I'll be unavailable until Jan 2nd 2025 so if there is any PR related to this issue, I let you in full control of merging it!

This "bug" is simply related to the fact that we don't search for nested structure tags. This can be fixed quickly when I come home in 6 days.

In the meantime, I hope it's not a blocker for you. Validation should still be working even if it's not mentionned in the spec.

**Comment by dylanhitt** [comment]
Cool, I should hopefully have time tomorrow assuming travel goes as planned.

**Comment by dylanhitt** [comment]
Should be fixed once #319 is merged.
